src/kontrol/foundry.py

```diff
@@ -253,18 +253,14 @@ def _escape_brackets(regs: list[str]) -> list[str]:
     def matching_tests(self, tests: list[str]) -> list[str]:
         all_tests = self.all_tests
         all_non_tests = self.all_non_tests
-        matched_tests = set()
-        unfound_tests: list[str] = []
         tests = self._escape_brackets(tests)
-        for t in tests:
-            if not any(re.search(t, test) for test in (all_tests + all_non_tests)):
-                unfound_tests.append(t)
-        for test in all_tests:
-            if any(re.search(t, test) for t in tests):
-                matched_tests.add(test)
-        for test in all_non_tests:
-            if any(re.search(t, test) for t in tests):
-                matched_tests.add(test)
+        matched_tests = set()
+        unfound_tests = set(tests)
+        for test in tests:
+            for possible_match in all_tests + all_non_tests:
+                if re.search(test, possible_match):
```

Comment:
@tothtamas28 would it make sense to have a single `re.search` on a string that has `all_tests + all_non_tests` instead?

Reply:
Not really, for two reasons:
* `re.search` returns the first match. You could use `re.findall` instead, but that still requires  post-processing to correlate matches to tests`.
* if there's a single string, the match might span the section of multiple tests (if I understand the question correctly), which doesn't make sense.

Reply:
Thanks!